test/Spec.hs

```diff
@@ -7,19 +7,19 @@ main :: IO ()
 main = hspec $ do
   describe "detect language for path" $ do
     it "can detect Ruby by file extension" $
-      languageName <$> languageForPath "test.rb" `shouldBe` Just "Ruby"
+      languageName <$> languagesForPath "test.rb" `shouldBe` ["Ruby"]
 
     it "can detect Ruby by filename" $
-      languageName <$> languageForPath "Rakefile" `shouldBe` Just "Ruby"
+      languageName <$> languagesForPath "Rakefile" `shouldBe` ["Ruby"]
 
     it "Gemfile.lock is not Ruby" $
-      languageName <$> languageForPath "Gemfile.lock" `shouldBe` Nothing
+      languageName <$> languagesForPath "Gemfile.lock" `shouldBe` []
```

Comment:
Let's add one test where we get back multiple results (php and hack is a good one).